Added QR Code model deployment tutorial (#323)
luxonis/depthai-docs-website: source/pages/tutorials/deploying-custom-model.rst

> Color/Mono cameras on OAK camera will output frames in U8 data type (``[0,255]``). Models are usually trained with

Reviewer: ```suggestion
Color/Mono OAK cameras will output frames in U8 data type (``[0,255]``). Models are usually trained with
```

Author: OAK-D camera has 3 cameras; 2x mono and 1x color - which is what I meant here

We might get better results if we used `ObjectTracker node <https://docs.luxonis.com/projects/api/en/latest/components/nodes/object_tracker/>`__
to track faces, but that's out of the scope of this tutorial.

2. QR code detector
-------------------

This tutorial will focus around deploying the **WeChat** `QR code detector <https://github.com/opencv/opencv_zoo/tree/4fb591053ba1201c07c68929cc324787d5afaa6c/models/qrcode_wechatqrcode>`__.

> I have found this model while going through `OpenCV Model Zoo <https://github.com/opencv/opencv_zoo>`__. There are 2 models in

Reviewer: ```suggestion
I found this model while going through `OpenCV Model Zoo <https://github.com/opencv/opencv_zoo>`__. There are 2 models in
```

as it was designed to decode these standard SSD detection results.

.. figure:: /_static/images/tutorials/deploying-custom-model/detectionOutput.png

    NMS layer in the model

.. code-block:: python

    # Normalize the bounding box to frame resolution.
    # For example, [0.5, 0.5, 1, 1] bounding box on 300x300 frame
    # should return [150, 150, 300, 300]
    def frame_norm(frame, bbox):
        bbox[bbox < 0] = 0
        normVals = np.full(len(bbox), frame.shape[0])
        normVals[::2] = frame.shape[1]
        return (np.clip(np.array(bbox), 0, 1) * normVals).astype(int)

    # ...

    # Do the NN inference
    output = exec_net.infer(inputs={input_blob: image})
    print(output) # Print the output

    results = output['detection_output'].reshape(100, 7)
    for det in results:
        conf = det[2]
        bb = det[3:]
        bbox = frame_norm(og_img, bb)
        cv2.rectangle(og_img, (bbox[0], bbox[1]) , (bbox[2], bbox[3]), (255, 127, 0), 3)

    cv2.imshow('USB Camera', og_img)
    if cv2.waitKey(1) == ord('q'): break

> After trying a few MEAN/SCALE values, I have found that MEAN=0 and SCALE=255 works the best. We don't need to worry

Reviewer: ```suggestion
After trying a few MEAN/SCALE values, I found that MEAN=0 and SCALE=255 works the best. We don't need to worry
```

about color order as the model requires grayscale images.

.. figure:: /_static/images/tutorials/deploying-custom-model/qr-detection.jpg

    Success! Light blue bounding box around the QR code!

Testing accuracy degredation due to FP16 quantization
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Now let's try the model with FP16 precision instead of INT32. If you an OAK camera into your computer you can select

Reviewer: ```suggestion
Now let's try the model with FP16 precision instead of INT32. If you connect an OAK camera to your computer you can select
```

        def frameNorm(frame, bbox):
            normVals = np.full(len(bbox), frame.shape[0])
            normVals[::2] = frame.shape[1]
            return (np.clip(np.array(bbox), 0, 1) * normVals).astype(int)

        def displayFrame(name, frame):
            color = (255, 255, 0)
            for detection in detections:
                bbox = frameNorm(frame, (detection.xmin, detection.ymin, detection.xmax, detection.ymax))
                cv2.putText(frame, labelMap[detection.label], (bbox[0] + 10, bbox[1] + 20), cv2.FONT_HERSHEY_TRIPLEX, 0.5, color)
                cv2.putText(frame, f"{int(detection.confidence * 100)}%", (bbox[0] + 10, bbox[1] + 40), cv2.FONT_HERSHEY_TRIPLEX, 0.5, color)
                cv2.rectangle(frame, (bbox[0], bbox[1]), (bbox[2], bbox[3]), color, 2)
            cv2.imshow(name, frame)

        while True:
            frame = qRight.get().getCvFrame()
            frame =  cv2.cvtColor(frame, cv2.COLOR_GRAY2BGR) # For colored visualization
            detections = inDet = qDet.get().detections
            displayFrame("right", frame)

            if cv2.waitKey(1) == ord('q'):
                break

QR Code end result
^^^^^^^^^^^^^^^^^^

> This is the end result of the script above. You can see that the mono camera sensor on OAK camera performs much better 

Reviewer: ```suggestion
This is the end result of the script above. You can see that the mono camera sensor on OAK cameras performs much better 
```
